test/model.populate.test.js

```diff
@@ -6073,5 +6073,87 @@ describe('model: populate:', function() {
         });
       });
     });
+    describe('populates an array of objects (gh6284)', function() {
+      it('with mulitiple space separated paths', function(done) {
+        var db = start();
+        var houseSchema = new Schema({ location: String });
+        var citySchema = new Schema({ name: String });
+        var districtSchema = new Schema({ name: String });
+
+        var userSchema = new Schema({
+          name: String,
+          houseId: {
+            type: Schema.Types.ObjectId,
+            ref: 'gh6284_0'
+          },
+          cityId: {
+            type: Schema.Types.ObjectId,
+            ref: 'gh6284_2'
+          },
+          districtId: {
+            type: Schema.Types.ObjectId,
+            ref: 'gh6284_3'
+          }
+        });
+
+        var postSchema = new Schema({
+          content: String,
+          userId: {
+            type: Schema.Types.ObjectId,
+            ref: 'gh6284_1'
+          }
+        });
+
+        var House = db.model('gh6284_0', houseSchema);
+        var User = db.model('gh6284_1', userSchema);
+        var City = db.model('gh6284_2', citySchema);
+        var District = db.model('gh6284_3', districtSchema);
+        var Post = db.model('gh6284_4', postSchema);
+
+        var house = new House({ location: '123 abc st.' });
+        var city = new City({ name: 'Some City' });
+        var district = new District({ name: 'That District' });
+
+        var user = new User({
+          name: 'Billy',
+          houseId: house._id,
+          districtId: district._id,
+          cityId: city._id
+        });
+
+        var post = new Post({
+          content: 'Some meaningful insight.',
+          userId: user._id
+        });
+
+        House.create(house, function() {
```

Comment:
Can you just use `co` here? We're trying to avoid callbacks in test code because co makes things much more readable.

Reply:
I'll checkout the existing examples of co and read the docs for it if I need to. I see a lot of tests use promises as well, is that also an acceptable option or should I stick with co?

Reply:
co is way better. it's like async/await but from 2 years before it existed :). done.